openedx/core/djangoapps/credit/tests/test_models.py

```diff
@@ -29,6 +31,19 @@ def test_is_credit_course(self, is_credit):
         else:
             self.assertFalse(CreditCourse.is_credit_course(self.course_key))
 
+    def test_cache_is_credit_course(self):
+        self.assertFalse(CreditCourse.is_credit_course(self.course_key))
+        CreditCourse(course_key=self.course_key, enabled=True).save()
+        cache.clear()
+
+        # Warm the cache
+        with self.assertNumQueries(1):
+            self.assertTrue(CreditCourse.is_credit_course(self.course_key))
+
+        # Load credit courses from cache
```

Comment:
`Test that for the second time, checking whether a course is credit course or not don't require querying the database, instead credit course data is fetched from cache`